### PR TITLE
Codegen support

### PR DESCRIPTION
**Pull request by bakhtiyarneyman**
This PR enables copying source files into the `src` directory before invoking `purs`. The usecase is to support packages like `graphql-client` which rely on generating the Purescript code. Committing such code to the codebase is an option that comes at the cost of making diffs and code reviews more difficult, so some teams opt for generating these files on the fly.

### CODE REVIEW

File: nix/examples/simple-codegen/src/Main.purs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module Example.Simple.Main where
+
+import Generated (answer)
```
Comment (thomashoneyman) [inline nix/examples/simple-codegen/src/Main.purs]:
```suggestion
import Generated (answer)

```

**Comment by thomashoneyman** [comment]
There is also an 'include' / 'exclude' pair available in the spago.yaml config:
https://github.com/purescript/spago?tab=readme-ov-file#the-configuration-file

However, it looks like it only applies to 'publish' so wouldn't be suitable during a build.

File: nix/examples/simple-codegen/src/Main.purs (hunk 1 above)
Comment (thomashoneyman) [reply nix/examples/simple-codegen/src/Main.purs]: We should ensure newlines even in generated files. Other than this :+1:

Comment (bakhtiyarneyman) [reply nix/examples/simple-codegen/src/Main.purs]: Done!

File: .vscode/settings.json
Comment (bakhtiyarneyman) [inline .vscode/settings.json:1]: @thomashoneyman Tagging to make sure checking this line doesn't fall through the cracks.

Comment (thomashoneyman) [reply .vscode/settings.json:1]: I’d rather not commit a .vscode directory, but otherwise 👍 on the pr! Thanks!

Comment (bakhtiyarneyman) [reply .vscode/settings.json:1]: Knew it was going to be contentious 😅 Done!

**Pull request merged**